publish/tasks.py

```diff
@@ -150,20 +153,26 @@ def _load_folder(client: Client, path: str, parent_id: str) -> List[DandiFile]:
     def s3_path(self, version):
         return f'{S3_PREFIX}/{self.dandi_id}/{version}'
 
-    def _get_next_version(self, s3: 'S3Client') -> int:
-        version = 1
-        while True:
-            try:
-                s3.get_object(Bucket=S3_BUCKET,
-                              Key=f'{self.s3_path(version)}/dandiset.yaml')
-            except ClientError as e:
-                if e.response['Error']['Code'] == 'NoSuchKey':
-                    return version
-                raise
-            version += 1
+    def _get_version_for_datetime(self, time):
+        return time.strftime('0.%y%m%d.%H%M')
+
+    def _get_next_version(self) -> str:
+        time = datetime.datetime.utcnow()
+        version = self._get_version_for_datetime(time)
+        # increment time until there are no collisions
+        collisions = models.Dandiset.objects\
+            .filter(dandi_id=self.dandi_id, version=version)\
+            .count()
```

Comment:
`exists()` is preferred over `count() > 0`: https://docs.djangoproject.com/en/3.0/ref/models/querysets/#django.db.models.query.QuerySet.exists